experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+                errmsg = "intt must be an integer"
+                raise ExperimentException(errmsg, exp_slice)
+            else:
+                try:
+                    assert 'intn' in exp_slice.keys()
+                except AssertionError:
+                    pass
+                else:
+                    # TODO Log warning intn will not be used
+                    exp_slice.pop('intn')
+
+        try:
+            assert 'beam_angle' in exp_slice.keys()
+            assert isinstance(exp_slice['beam_angle'], list)
+            for element in exp_slice['beam_angle']:
+                assert isinstance(element, float) or isinstance(element, int) or isinstance(element,
+                                                                                            list)
+                if isinstance(element, list):
+                    for angle in element:
+                        assert isinstance(angle, float) or isinstance(angle, int)
+        except AssertionError:
+            errmsg = "Slice must specify beam_angle that must be a list of floats or lists (of floats)"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'beam_order' in exp_slice.keys()
+            assert isinstance(exp_slice['beam_order'], list)
+            for element in exp_slice['beam_order']:
+                assert isinstance(element, int)
+                assert element < len(exp_slice['beam_angle'])
+        except AssertionError:
+            errmsg = "Slice must specify beam_order that must be a list of ints corresponding to " \
+                     "the order of the angles in the beam_angle list."
+            raise ExperimentException(errmsg, exp_slice)
+
+    @staticmethod
+    def set_slice_identifiers(exp_slice):
+        """
+        This function sets up internal identifier flags 'clrfrqflag' and 'rxonly' in the slice so 
+        that we know how to properly set up the slice and know which keys in the slice must be 
+        specified and which are unnecessary. If these keys are ever written by the user, they will 
+        be rewritten here.
+        :param exp_slice: 
+        :return: 
+        """
+
+        if 'clrfrqrange' in exp_slice.keys():
+
+            exp_slice['clrfrqflag'] = True
+            exp_slice['rxonly'] = False
+
+            txfreq = exp_slice.pop('txfreq', None)
+            if txfreq is not None and txfreq not in \
+                    range(exp_slice['clrfrqrange'][0],
+                          exp_slice['clrfrqrange'][1]):
+                pass  # TODO log a warning. Txfreq is removed but we may not be doing as you intended.
+
+            rxfreq = exp_slice.pop('rxfreq', None)
+            if rxfreq is not None and rxfreq not in \
+                    range(exp_slice['clrfrqrange'][0],
+                          exp_slice['clrfrqrange'][1]):
+                pass  # TODO log a warning. Rxfreq is removed but we may not be doing as you intended.
```

Comment:
rxfreq is removed here (previous line exp_slice.pop()). The rxfreq will = txfreq here. If you want a different rxfreq you should add another slice. The warning is in case someone tries to add a separate rxfreq from txfreq or clrfrqrange inside the same slice.